### ISSUE TITLE
Build fails with clang: unsupported: typedef changes linkage of anonymous type, but linkage was already computed

### ISSUE THREAD

**Issue by yurivict** [falkTX/Carla#651]
I'm getting these errors/warnings:
```
In file included from ../CarlaPlugin.hpp:21:
../CarlaBackend.h:1467:3: error: unsupported: typedef changes linkage of anonymous type, but linkage was already computed
} ParameterRanges;
  ^
../CarlaBackend.h:1338:15: note: use a tag name here to establish linkage prior to definition
typedef struct {
              ^
               ParameterRanges
In file included from CarlaEngine.cpp:25:
In file included from ./CarlaEngineInternal.hpp:21:
In file included from ./CarlaEngineOsc.hpp:25:
../CarlaBackend.h:1467:3: error: unsupported: typedef changes linkage of anonymous type, but linkage was already computed
} ParameterRanges;
  ^
../CarlaBackend.h:1338:15: note: use a tag name here to establish linkage prior to definition
typedef struct {
              ^
               ParameterRanges
```

Revision: a8c5ddd1

**Comment by falkTX** [comment]
First time I see this. I guess just adding something like:
`typedef struct _ParameterRanges {`
would fix it.
Can you confirm?

**Comment by yurivict** [comment]
Yes, this helped.

**Comment by falkTX** [comment]
Cool, please make the other needed changes and send a PR?
Thanks,

**Comment by falkTX** [comment]
Fixed in #652, thanks!
